tools/deploy-cli/templates/ironic_bmo_configmap_env.tpl

```diff
@@ -0,0 +1,16 @@
+HTTP_PORT=6180
+{{ if .DeployKeepAlived }}
+PROVISIONING_IP=172.22.0.2
+PROVISIONING_INTERFACE=ironicendpoint
+{{ else }}
+PROVISIONING_INTERFACE=eth2
+IRONIC_INSPECTOR_VLAN_INTERFACES=all
+{{ end }}
+DHCP_RANGE=172.22.0.10,172.22.0.100
+DEPLOY_KERNEL_URL=http://172.22.0.2:6180/images/ironic-python-agent.kernel
+DEPLOY_RAMDISK_URL=http://172.22.0.2:6180/images/ironic-python-agent.initramfs
+IRONIC_ENDPOINT=http://172.22.0.2:6385/v1/
+CACHEURL=http://172.22.0.1/images
```

Comment:
I don't think we can have these hard coded. It will only work in CI. I think the best solution is to provide a flag (e.g. `--ironic-env-file`) that can be used to give the whole file instead of templating. We can have some default template but it will never be able to support all possible configurations.

Reply:
I don't have an opinion about it, but imo this would live together just fine with the said `--ironic-env-file` flag. (We can use this one only for CI, for e.g.). This hardcoded template simply is what `deploy.sh` does.

Reply:
Yes it is what deploy.sh does, but it is actually not what we use in CI. We have an [ugly hack in metal3-dev-env](https://github.com/metal3-io/metal3-dev-env/blob/af8d1765c7d66640377ba7632419b3c3409ce018/03_launch_mgmt_cluster.sh#L142-L162) where we replace this file before we run the deploy script. :see_no_evil: 
With the template and hard-coded values built in to the binary this will no longer be possible. So I think we must at least add a flag to override and preferably also allow overriding each of the values with environment variables.
Is it possible to do the equivalent of `DHCP_RANGE=${DHCP_RANGE:-172.22.0.10,172.22.0.100}`?

Reply:
Yes, defaults would be very nice!